docs/upgrade.md

```diff
@@ -3,5 +3,261 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+We've made targeted breaking changes to make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+| Area                                   | Change                                                                                                                                           | Code change required |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | No                   |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
+| **Types imports**                      | Updated import path for TypeScript types to leverage subpath exports - i.e. `@aws-lambda-powertools/logger/types`.                               | Yes                  |
+| **Logger - log sampling**              | Changed implementation of [log sampling](./core/logger.md#sampling-logs) to dynamically switch log level to `DEBUG` on a percentage of requests. | -                    |
```

Comment:
Should this be a "No" in a last column or is it not clear and require more explanation to narrow down the cases?

Reply:
I changed the other one (first line) to `-` as well, so they're the same.

Reply:
What does `-` mean?

Reply:
It means `No`. 

I decided to follow what Python did with their upgrade guide and use `-` as well in an effort to keep the info density low (aka the table looks less dense).

But if we think it's unclear or confusing let's put `No` directly.

Reply:
The `-` keeps customers visually focused on what needs action.

Reply:
I still don't understand why we have two different semantics for a `No` in a Yes/No column.